From bdeca3f4692fecec66f776b39926d9e1ee0c369b Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Wed, 8 Dec 2021 23:50:48 +0200
Subject: [PATCH 4/6] drivers: mmc: Fix uninitialized variable warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

If we compile with -O3, we get an
  include/drivers/mmc.h:93:33: error: ‘resp_data’ may be used uninitialized
  in this function [-Werror=maybe-uninitialized]
warning.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-8268
---
 drivers/mmc/mmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/mmc.c b/drivers/mmc/mmc.c
index e288d47ed..26dcd1711 100644
--- a/drivers/mmc/mmc.c
+++ b/drivers/mmc/mmc.c
@@ -77,7 +77,7 @@ static int mmc_send_cmd(unsigned int idx, unsigned int arg,
 static int mmc_device_state(void)
 {
 	int retries = MMC_DEFAULT_MAX_RETRIES;
-	unsigned int resp_data[4];
+	unsigned int resp_data[4] = {0};
 
 	do {
 		int ret;
-- 
2.17.1

